### PR TITLE
produce a string for the front end to ingest

### PR DESCRIPTION
**Pull request by emma-nechamkin**

### CODE REVIEW

**Comment by emma-nechamkin** [comment]
I have no idea why all of these commits show up??

File: data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py [hunk 1]
```diff
@@ -68,6 +69,42 @@ def _create_string_from_list(series: pd.Series) -> str:
...
+        What are these rules?
+        0. If None, return none
+        1. If the percentage is below 1%, produce 'less than 1%'
+        2. If the percentage is above 99.95%, produce '100%'
+        3. If the percentage is X.00 when rounded to two sig digits, display the integer of the percent
+        4. If the percentage has unique significant digits, report two digits
+        """
+        # Rule 0
+        if not percentage_float:
+            # I believe we need to do this because JS will do weird things with a mix-type column?
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py:85]: nice comment

File: data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py [hunk 2]
```diff
@@ -68,6 +69,42 @@ def _create_string_from_list(series: pd.Series) -> str:
...
+        # Rule 3 (this is a shorthand because round(4) will truncate repeated 0s)
+        if last_digits[-1] == "0":
+            return first_digits + "%"
+
+        # Rule 4
+        if last_digits != "00":
+            return rounded_percentage_str + "%"
+
+        # There is something missing!
+        raise Exception("Yikes! The string conversion here failed!")
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py:106]: nice

File: data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py [hunk 3]
```diff
@@ -209,4 +246,12 @@ def transform(self) -> None:
             merged_output_df[field_names.PERCENT_OF_TRIBAL_AREA_IN_TRACT],
         )
 
+        # The very final thing we want to do is produce a string for the front end to show
+        # We do this here so that all of the logic is included
+        merged_output_df[
+            field_names.PERCENT_OF_TRIBAL_AREA_IN_TRACT_DISPLAY_STRING
+        ] = merged_output_df[field_names.PERCENT_OF_TRIBAL_AREA_IN_TRACT].apply(
+            self._adjust_percentage_to_string
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/tribal_overlap/etl.py:254]: 🎉

**Review by lucasmbrown-usds — APPROVED** [review]
Looks great!

**Comment by emma-nechamkin** [comment]
(this is in theory ready to be merged once the checks pass and will have new strings for @vim-usds to ingest on the front end)

**Pull request merged**